### DISCUSSION TITLE
Composer require fails with Undefined array key "database_dump_filename_base"

### DISCUSSION THREAD

**Discussion by agustinzamar** [spatie/laravel-backup#1824] [General]
Simply running `composer require spatie/laravel-backup` or `composer update` after changing the version from 8.* to 9.* will fail if the config/backup.php file is present.

![image](https://github.com/user-attachments/assets/74b46074-7f11-420a-abc0-e3ae3815367b)

**Comment by Nielsvanpach** [comment]
I'd suggest to keep your config file in line with the one in v9:
https://github.com/spatie/laravel-backup/blob/main/config/backup.php

**Reply by BulletGoof** [reply]
I have the same error and copying the config file from the v9 version does not work.

**Reply by Nielsvanpach** [reply]
Are you sure you have this line in your config file?
https://github.com/spatie/laravel-backup/blob/main/config/backup.php#L108

**Reply by BulletGoof** [reply]
Yes I have that exact line in it.

This is a command and output I receive, the command can vary but the output remains the same:
PS C:\inetpub\wwwroot\snipe-it> php artisan db

   ErrorException

  Undefined array key "database_dump_filename_base"

  at vendor\spatie\laravel-backup\src\Config\BackupConfig.php:40
     36▕             name: $data['name'],
     37▕             source: SourceConfig::fromArray($data['source']),
     38▕             databaseDumpCompressor: $data['database_dump_compressor'] ?? null,
     39▕             databaseDumpFileTimestampFormat: $data['database_dump_file_timestamp_format'] ?? null,
  ➜  40▕             databaseDumpFilenameBase: $data['database_dump_filename_base'],
     41▕             databaseDumpFileExtension: $data['database_dump_file_extension'],
     42▕             destination: DestinationConfig::fromArray($data['destination']),
     43▕             temporaryDirectory: $data['temporary_directory'] ?? null,
     44▕             password: $data['password'],

  1   vendor\spatie\laravel-backup\src\Config\BackupConfig.php:40
      Illuminate\Foundation\Bootstrap\HandleExceptions::Illuminate\Foundation\Bootstrap\{closure}()

  2   vendor\spatie\laravel-backup\src\Config\Config.php:28
      Spatie\Backup\Config\BackupConfig::fromArray()

**Reply by Nielsvanpach** [reply]
Is it set to NULL? Or cached? BackupConfig only reads out your `backups.php` config file and let's you know the `database_dump_filename_base` key is missing a value.

**Reply by BulletGoof** [reply]
it is set to the following:
        /*
         * The base of the dump filename, either 'database' or 'connection'
         *
         * If 'database' (default), the dumped filename will contain the database name.
         * If 'connection', the dumped filename will contain the connection name.
         */
        'database_dump_filename_base' => 'database',

it's used in snipe-it and there you have an .env file where the DB connection is set in.
I tried setting the line 80 to the database itself instead of mysql but that does not change it either.

This worked before the upgrade that happened this morning but the upgrade is now half stuck because of this error message. 

If you want I can add some files that you need?

**Comment by tigusigalpa** [comment]
if you have Laravel version under 11, you have to paste `composer require spatie/laravel-backup:^8.0`
